File: dart/darts/lib/darts.dart
import 'dart:math';

class Darts {
  int score(double x, double y) {
    double distance = sqrt(x * x + y * y);

    if (distance <= 1) {
      return 10;
    } else if (distance <= 5) {
      return 5;
    } else if (distance <= 10) {
      return 1;
    }
    return 0;
  }
}

File: clojure/darts/src/darts.clj
(ns darts
  (:require [clojure.math :as math]))

(defn distance [x y] (math/sqrt (+ (math/pow x 2) (math/pow y 2))))

(defn score [x y]
  (condp < (distance x y)
    10 0
    5 1
    1 5
    10))
